### PR TITLE
Attempt at getting better layout by adding some spans

### PR DESCRIPTION
**Pull request by Drup**
This is not perfect by a long shot, but it's marginally better than before.
The diff is ridiculously large because it adds lot's of extra spans.

@lpw25 If you could test that on your problematic rendering, that would be good.

### CODE REVIEW

File: test/html/expect/test_package+custom_theme,ml/Include/index.html [hunk 1]
```diff
@@ -19,68 +19,68 @@
...
     <div class="spec include">
      <div class="doc">
       <details open="open">
        <summary>
-        <span class="def"><code><span class="keyword">include</span> <a href="module-type-Not_inlined/index.html">Not_inlined</a></code></span>
+        <span class="def"><code><span><span class="keyword">include</span> <a href="module-type-Not_inlined/index.html">Not_inlined</a></span></code></span>
        </summary>
        <div class="spec type" id="type-t">
-        <a href="#type-t" class="anchor"></a><code><span class="keyword">type</span> t</code>
+        <a href="#type-t" class="anchor"></a><code><span><span class="keyword">type</span> t</span><span></span><span></span><span></span></code>
```
Comment (Octachron) [inline test/html/expect/test_package+custom_theme,ml/Include/index.html:37]: Is it possible to elide those empty spans?

Comment (Drup) [reply test/html/expect/test_package+custom_theme,ml/Include/index.html]: Should be done now. maybe even too aggressively (it removes empty code blocks too...)

**Comment by dbuenzli** [comment]
@Drup could you give a rough idea of what you did ? I was thinking we should maybe try to find a new rendering strategy (separation of content and styling is a myth).

We tried [two](https://github.com/ocaml/odoc/issues/19) so far. I think the last one with `inline-block` `span`s generally does a good job (I don't think `odoc`'s stylesheet does it) but sadly it [breaks](https://github.com/ocaml/odoc/issues/125#issuecomment-504468736) in-page search on Chrome, the search doesn't work across the `inline-block`ed spans.

**Comment by Drup** [comment]
@dbuenzli Ah, I see I was not precise enough: I just added back some of the spans that disappeared in the turmoil. Nothing else. This is only about browser-indentation, to make it less horrible than before.

**Comment by dbuenzli** [comment]
> We tried [two](https://github.com/ocaml/odoc/issues/19) so far. I think the last one with `inline-block` `span`s generally does a good job (I don't think `odoc`'s stylesheet does it) but sadly it [breaks](https://github.com/ocaml/odoc/issues/125#issuecomment-504468736) in-page search on Chrome, the search doesn't work across the `inline-block`ed spans.

Just FTR I found a way to fix that see [here](https://github.com/ocaml/odoc/issues/125#issuecomment-770105925).

**Comment by jonludlam** [comment]
I've rebased and fixed the tests on old OCaml versions in #604 so closing this one.

**Comment by dbuenzli** [comment]
@Drup honestly I dont really see what this brought to the party (and FWIW, it broke, the search wise but visual wise excellent `display: inline-block` approach). This rather seem to have derived into odd `span`itis. For example having functions arrows are inside a span with their left argument which seems quite odd.

One thing I now realized is that `&nbsp` does seem c&p as space. So that's something we should play with to avoid odd cuts. E.g. between type parameters and type constructors.

**Pull request closed without merging**